### PR TITLE
fix: thorchain LP evm fees estimation

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Fixes two issues with EVM Txs fee estimation for THORChain LP deposit:

- One with missed base unit / crypto precision conversion
- Another with trying to estimate token deposits - we were assuming token *sends* (i.e `contractAddress` being defined) but at the same time doing a contract call, which produces invalid hex `data`. 

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

Low, this *fixes* fee estimation and cannot break things more as they were obviously wrong

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

- Fees estimation is happy for THORChain EVM *token* deposits at confirm step
- Fees estimation is happy for THORChain EVM *native assets* deposits at confirm step 

### Engineering

- ☝🏽 

### Operations

- ☝🏽 

## Screenshots (if applicable)

Screenshot 2024-03-04 at 10 12 52
image

### CODE REVIEW

File: src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx [hunk 1]
```diff
@@ -307,15 +307,16 @@ export const TransactionRow: React.FC<TransactionRowProps> = ({
       case 'EvmCustomTx': {
         if (!inboundAddressData?.router) return undefined
         if (!accountAssetAddress) return undefined
-        const assetAddress = isToken(fromAssetId(assetId).assetReference)
-          ? getAddress(fromAssetId(assetId).assetReference)
-          : zeroAddress
-        const amountOrDustAmountCryptoBaseUnit = // Reuse the savers util as a sane amount for the dust threshold
-          BigInt(
+        const amountOrDustAmountCryptoBaseUnit = (() => {
```
Comment (kaladinlight) [inline src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:310]: we need two values here.

amount for `depositWithExpiry` is the value being sent (eth or token) for the tx

amount for tx and gas estimation should be the eth amount being deposited or dust amount being sent for withdraw

Comment (kaladinlight) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:310]: This value is correct for `amountCryptoPrecision` on the return args, I am saying we are constructing the memo incorrectly here now as in not passing the desired transfer amount (being zeroed out for tokens)

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx:310]: Oh I see what you meant now. Absolutely correct!

Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/ReusableLpStatus/TransactionRow.tsx]: da29670928268ddb06d858899e2b8d98f441b191 and 4c842ab5acc5160aa6312bf45aa0ed746d56622e

image
image
image


image
image
image

**Review by 0xApotheosis — APPROVED** [review]
This looks sane, and fixes the issues I was debugging last night.

I am getting some strange issues related to fee asset balance, but I don't think they are at all related to this PR. I have plenty of ETH in the screenshot below.

Screenshot 2024-03-05 at 9 37 30 am

I'll fix them in a separate PR.

**Review by gomesalexandre — COMMENTED** [review]
Retested after big @kaladinlight changes:

- Fees estimation is still happy for native EVM assets

image
image

- Txs and their call data is still happy for native EVM assets

image

- Fees estimation is still happy for EVM tokens

image
image

- Txs and their call data is still happy for tokens

image

**Review by kaladinlight — APPROVED** [review]

**Pull request merged**